data/npc/scripts/bank.lua

```diff
@@ -0,0 +1,347 @@
+local keywordHandler = KeywordHandler:new()
+local npcHandler = NpcHandler:new(keywordHandler)
+NpcSystem.parseParameters(npcHandler)
+
+local count = {}
+local transfer = {}
+
+function onCreatureAppear(cid)              npcHandler:onCreatureAppear(cid)            end
+function onCreatureDisappear(cid)           npcHandler:onCreatureDisappear(cid)         end
+function onCreatureSay(cid, type, msg)      npcHandler:onCreatureSay(cid, type, msg)    end
+function onThink()                          npcHandler:onThink()                        end
+
+local function greetCallback(cid)
+	count[cid], transfer[cid] = nil, nil
+	return true
+end
+
+local function creatureSayCallback(cid, type, msg)
+	if not npcHandler:isFocused(cid) then
+		return false
+	end
+	local player = Player(cid)
+	if msgcontains(msg, "bank account") then
+		npcHandler:say({
+			"Every citizen has one. The big advantage is that you can access your money in every branch of the World Bank! ...",
+			"Would you like to know more about the {basic} functions of your bank account, the {advanced} functions, or are you already bored, perhaps?"
+		}, cid)
+		npcHandler.topic[cid] = 0
+		return true
+	elseif msgcontains(msg, "balance") then
+		npcHandler.topic[cid] = 0
+		if player:getBankBalance() >= 100000000 then
+			npcHandler:say("I think you must be one of the richest inhabitants in the world! Your account balance is " .. player:getBankBalance() .. " gold.", cid)
+			return true
+		elseif player:getBankBalance() >= 10000000 then
+			npcHandler:say("You have made ten millions and it still grows! Your account balance is " .. player:getBankBalance() .. " gold.", cid)
+			return true
+		elseif player:getBankBalance() >= 1000000 then
+			npcHandler:say("Wow, you have reached the magic number of a million gp!!! Your account balance is " .. player:getBankBalance() .. " gold!", cid)
+			return true
+		elseif player:getBankBalance() >= 100000 then
+			npcHandler:say("You certainly have made a pretty penny. Your account balance is " .. player:getBankBalance() .. " gold.", cid)
+			return true
+		else
+			npcHandler:say("Your account balance is " .. player:getBankBalance() .. " gold.", cid)
+			return true
+		end
+	elseif msgcontains(msg, "deposit") then
+		count[cid] = player:getMoney()
+		if count[cid] < 1 then
+			npcHandler:say("You do not have enough gold.", cid)
```

Comment:
some lines repeat multiple times
shouldn't they be referenced to some local variable?